03_DeviceSelectionAndSharedSources/Testers.h

```diff
@@ -33,15 +33,18 @@ class IntrospectionTesterBase
 		// this time we load a shader directly from a file
 		auto assetBundle = assetMgr->getAsset(shaderPath, lp);
 		const auto assets = assetBundle.getContents();
-		if (assets.empty())
+		const auto* metadata = assetBundle.getMetadata();
+		if (assets.empty() || assetBundle.getAssetType() != IAsset::ET_SHADER)
 		{
 			logFail(logger, "Could not load shader!");
 			assert(0);
 		}
+		const auto hlslMetadata = static_cast<const CHLSLMetadata*>(metadata);
```

Comment:
before casting to a certain metadata type, you should check the metadata for being of that type